tests/ui/test_search.py

```diff
@@ -0,0 +1,42 @@
+import pytest
+
+from pages.desktop.home import Home
+
+
+@pytest.mark.nondestructive
+def test_that_page_with_search_results_has_correct_title(
+        my_base_url, selenium, initial_data):
+    """Test the search term entered loads on the search page."""
+    home_page = Home(selenium, my_base_url).open()
+    search_keyword = 'Search term'
+    home_page.search_for(search_keyword)
+    assert search_keyword in selenium.title
+
+
+@pytest.mark.smoke
+@pytest.mark.nondestructive
+def test_that_searching_for_addon_returns_addon_as_first_result(
+        my_base_url, selenium, addon):
+    """Test searching for an addon returns the addon."""
+    home_page = Home(selenium, my_base_url).open()
+    name = home_page.featured_extensions.extensions[0].name
+    search_page = home_page.search_for(name)
+    results = [result.name for result in search_page.results]
+    assert name == results[0]
+
+
+@pytest.mark.native
+@pytest.mark.nondestructive
+@pytest.mark.parametrize('sort_by', ['Most Users', 'Top Rated'])
+def test_sorting_by(my_base_url, selenium, addon, addon_2, sort_by):
+    """Test searching for an addon and sorting."""
+    home_page = Home(selenium, my_base_url).open()
+    name = home_page.featured_extensions.extensions[0].name
+    search_page = home_page.search_for(name)
+    search_page.sort_by(sort_by)
+    if sort_by == 'Most Users':
+        results = [i.users for i in search_page.results]
```

Comment:
We should avoid this logic and matching based on the parameters. Instead, you could have an additional parameter for the property to check. For example:

```python
@pytest.mark.parametrize('sort_by, sort_attr', [['Most Users', 'users'], ['Top Rated', 'rating']])
def test_sorting_by(my_base_url, selenium, addon, addon_2, sort_by, sort_attr):
...
search_page.sort_by(category)
results = [getattr(i, sort_attr) for i in search_page.results]
assert sorted(results, reverse=True) == results
```